htlcswitch/switch.go

```diff
@@ -1364,7 +1442,19 @@ func (s *Switch) closeCircuit(pkt *htlcPacket) (*PaymentCircuit, error) {
 
 	// Otherwise, this is packet was received from the remote party.  Use
 	// circuit map to find the incoming link to receive the settle/fail.
-	circuit, err := s.circuits.CloseCircuit(pkt.outKey())
+	var circuit *PaymentCircuit
+	var err error
+	if pkt.circuitInKey != nil {
+		// In case we got the circuitInKey populated it means that we should
+		// use it to find the circuit as there is no open circuit probably due
+		// to interception of this packet.
+		circuit = s.circuits.LookupCircuit(*pkt.circuitInKey)
```

Comment:
is this circuit ever deleted?

Reply:
Yes, this test verifies that for all resolve options: https://github.com/lightningnetwork/lnd/pull/4018/files#diff-9ca6c00aa4cbaaa08aaa8bd697703398R3115